python/lsst/meas/extensions/shapeHSM/_hsm_moments.py

```diff
@@ -0,0 +1,548 @@
+            afwTable.CoordinateType.PIXEL,
+        )
+        if config.addFlux:
+            self.fluxKey = schema.addField(schema.join(name, "Flux"), type=float, doc="HSM PSF flux")
+
+    def _measure(
+        self,
+        record,
+        exposure,
+        debiasedPsfMoments=False,
+        noiseSource=None,
+        seedOffset=None,
+        badMaskPlanes=None,
+    ):
+        """
+        Measure adaptive moments of the PSF given an exposure and set the
+        results in the record in place.
+
+        Parameters
+        ----------
+        record : `~lsst.afw.table.SourceRecord`
+            The record where measurement outputs will be stored.
+        exposure : `~lsst.afw.image.Exposure`
+            The exposure containing the PSF which needs measurement.
+        debiasedPsfMoments : `bool`, optional
+            Debias the PSF moments if True.
+        noiseSource : `str`, optional
+            Noise source. How to choose variance of the zero-mean Gaussian
+            noise added to image? Allowed values are "meta" and "variance".
+            Becomes relevant only if `debiasedPsfMoments` is True.
+        seedOffset : `int`, optional
+            The seed offset for random number generator. Becomes relevant only
+            if `debiasedPsfMoments` is True.
+
+        Raises
+        ------
+        MeasurementError
+            Raised for errors in measurement.
+        """
+        # Extract the centroid from the record.
+        center = self.centroidExtractor(record, self.flagHandler)
+
+        # Retrieve the PSF from the exposure.
+        psf = exposure.getPsf()
+
+        # Check that the PSF is not None.
+        if not psf:
+            raise measBase.MeasurementError(self.NO_PSF.doc, self.NO_PSF.number)
+
+        # Two methods for getting PSF image evaluated at the source centroid:
+        if self.config.useSourceCentroidOffset:
+            # 1. Using `computeImage()` returns an image in the same coordinate
+            # system as the pixelized image.
+            psfImage = psf.computeImage(center)
+        else:
+            psfImage = psf.computeKernelImage(center)
+            # 2. Using `computeKernelImage()` to return an image does not
+            # retain any information about the origial bounding box of the
+            # PSF. We therefore reset the origin to be the same as the
+            # pixelized image.
+            psfImageBBoxAtCenter = psf.computeImageBBox(center)
+            psfImage.setXY0(psfImageBBoxAtCenter.getMin())
+
+        # Get the trace radius of the PSF.
+        psfSigma = psf.computeShape(center).getTraceRadius()
+
+        # Get the bounding box in the parent coordinate system.
+        bbox = psfImage.getBBox(afwImage.PARENT)
+
+        # Turn bounding box corners into GalSim bounds.
+        xmin, xmax = bbox.getMinX(), bbox.getMaxX()
+        ymin, ymax = bbox.getMinY(), bbox.getMaxY()
+        bounds = galsim.bounds.BoundsI(xmin, xmax, ymin, ymax)
+
```

Comment:
For the record, I'm not a big fan of this method doing the debiased version and the regular version based on `debiasedPsfMoments`, but it will do for now.

Reply:
Yeah, I think we ought to address this now.  One possibility may be to put the logic under the `if self._debiased` block into an overridden method `HsmPsfMomentsDebiasedPlugin.adjust_noise()`, which could just be a noop in `HsmPsfMomentsPlugin`.  That'll keep the logic of debiasing within the Debiased plugin, and I think it'll enable you to get rid of the `_debiased` class variable too.

Reply:
Good idea. Implemented!